lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -192,37 +191,45 @@ internal class EvaluatingCompiler(
             }
         }
 
-        override fun compute() = current?.exprValue() ?: valueFactory.nullValue
+        override fun compute() = current ?: valueFactory.nullValue
     }
 
-    private fun comparisonAccumulator(cmpFunc: (Number, Number) -> Boolean): (Number?, ExprValue) -> Number =
-        { curr, next ->
-            val nextNum = next.numberValue()
-            when (curr) {
-                null -> nextNum
-                else -> when {
-                    cmpFunc(nextNum, curr) -> nextNum
-                    else -> curr
-                }
+    private fun comparisonAccumulator(comparator: NaturalExprValueComparators): (ExprValue?, ExprValue) -> ExprValue =
+        { left, right ->
+            when {
+                left == null || comparator.compare(left, right) > 0 -> right
+                else -> left
             }
         }
 
     /** Dispatch table for built-in aggregate functions. */
     private val builtinAggregates: Map<Pair<String, PartiqlAst.SetQuantifier>, ExprAggregatorFactory> =
         run {
-            val countAccFunc: (Number?, ExprValue) -> Number = { curr, _ -> curr!! + 1L }
-            val sumAccFunc: (Number?, ExprValue) -> Number = { curr, next ->
-                curr?.let { it + next.numberValue() } ?: next.numberValue()
+            fun checkIsNumberType(funcName: String, value: ExprValue) {
+                if (!value.type.isNumber) {
+                    throw errNoContext(
```

Comment:
How is `throw` even allowed here?  I am not sure at all why this even builds, considering that `errNoContext` returns `Nothing`.  I'm surprised there's no compile error here.... actually, having tested this, it does generate a compile warning.  Please remove `throw`.